stdlib/ballerina-mime/src/main/ballerina/ballerina/mime/natives.bal

```diff
@@ -277,3 +277,141 @@ public const string DEFAULT_CHARSET = "UTF-8";
 
 @Description {value:"Permission to be used with opening a byte channel for overflow data"}
 const string READ_PERMISSION = "r";
+
+@Description {value:"Represent 'content-length' header name"}
+public const string CONTENT_LENGTH = "content-length";
+
+@Description {value:"Represent 'content-type' header name"}
+public const string CONTENT_TYPE = "content-type";
+
+@Description {value:"Get the header value associated with the given header name"}
+@Param {value:"entity: Represent the MIME entity"}
+@Param {value:"headerName: Represent header name"}
+@Return {value:"Return header value associated with the given header name. If multiple header values are present,
+then the first value will be returned"}
+public function <Entity entity> getHeader (string headerName) (string) {
+    if (entity.headers == null) {
+        return null;
+    }
+    var headerValue, _ = (string[])entity.headers[headerName];
+    if (headerValue != null) {
+        return headerValue[0];
+    } else {
+        //Ignore case and check existence of header
+        string[] headerValueArray = useCaseInsensitiveStrategy(headerName, entity.headers);
+        return headerValueArray == null ? null : headerValueArray[0];
+    }
+}
+
+@Description {value:"Get all the header values associated with the given header name"}
+@Param {value:"entity: Represent a MIME entity"}
+@Param {value:"headerName: Represent the header name"}
+@Return {value:"Return all the header values associated with the given header name as a string of arrays"}
+public function <Entity entity> getHeaders (string headerName) (string[]) {
+    if (entity.headers == null) {
+        return null;
+    }
+    return useCaseInsensitiveStrategy(headerName, entity.headers);
+}
+
+@Description {value:"Get all the headers"}
+@Param {value:"entity: Represent a MIME entity"}
+@Return {value:"Return all the headers in the entity"}
+public function <Entity entity> getAllHeaders () (map) {
+    if (entity.headers == null) {
+        return null;
+    }
+    return entity.headers;
+}
+
+@Description {value:"Add the given header value against the given header"}
+@Param {value:"entity: Represent a MIME entity"}
+@Param {value:"headerName: Represent the header name"}
+@Param {value:"headerValue: Represent the header value to be added"}
+public function <Entity entity> addHeader (string headerName, string headerValue) {
+    if (entity.headers == null) {
+        entity.headers = {};
+    }
+    string[] existingValues = useCaseInsensitiveStrategy(headerName, entity.headers);
+    if (existingValues == null) {
+        entity.setHeader(headerName, headerValue);
+    } else {
+        existingValues[lengthof existingValues] = headerValue;
+        string existingHeaderName = getCaseInsensitiveHeaderName(headerName, entity.headers);
```

Comment:
What's the need for looking up an existing header name here? If we do that, basically we are changing the header name the user set isn't it?

Reply:
if a header exists regardless of its case, we add the value to the same header. User can't change the header name if it exists already unless he does a setHeader